packages/notifications/src/styled/StyledIcon.spec.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import { Hue } from 'styled-components';
+import { render, renderRtl } from 'garden-test-utils';
+import { DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import { VALIDATION_TYPE, ARRAY_VALIDATION_TYPE } from '../utils/types';
+import { StyledIcon } from './StyledIcon';
+
+describe('StyledIcon', () => {
+  it('should render with the correct styling for each validation type', () => {
+    const { space, colors, palette } = DEFAULT_THEME;
+    const { successHue, dangerHue, warningHue, neutralHue } = colors;
+    const typeHues: Record<VALIDATION_TYPE, Hue> = {
+      success: palette[successHue][600],
+      error: palette[dangerHue][600],
+      warning: palette[warningHue][700],
+      info: palette[neutralHue][600]
+    };
+
+    ARRAY_VALIDATION_TYPE.forEach(type => {
+      const { container } = render(<StyledIcon type={type} />);
+
+      expect(container.firstChild).toHaveStyleRule('position', 'absolute');
+      expect(container.firstChild).toHaveStyleRule('margin-top', `${space.base / 2}px`);
+      expect(container.firstChild).toHaveStyleRule('color', typeHues[type] as string);
+    });
+  });
+
+  it('should render with the correct styling for RTL writing systems', () => {
+    const { container } = renderRtl(<StyledIcon type="success" />);
+    const { space } = DEFAULT_THEME;
+
+    expect(container.firstChild).toHaveStyleRule('right', `${space.base * 6}px`);
+    expect(container.firstChild).not.toHaveStyleRule('left');
+  });
+
+  it('should render with the correct styling for LTR writing systems', () => {
+    const { container } = render(<StyledIcon type="info" />);
+    const { space } = DEFAULT_THEME;
+
+    expect(container.firstChild).toHaveStyleRule('left', `${space.base * 6}px`);
+    expect(container.firstChild).not.toHaveStyleRule('right');
+  });
+});
```

Comment:
Unfortunately, I wasn't able to come up with tests that verify that the component renders the correct SVG icon depending on the `type`. I will continue to think about a test for verifying the correct SVG icon.